packages/edit-site/src/components/global-styles/utils.js

```diff
@@ -11,6 +11,30 @@ export function getVariationClassName( variation ) {
 	return `is-style-${ variation }`;
 }
 
+/**
+ * Returns a new index based on the presets and slug prefix.
+ *
+ * @param {Array}  presets    The array of presets.
+ * @param {string} slugPrefix The prefix for the slug.
+ *
+ * @return {number} The new index.
+ */
+export function getNewIndexFromPresets( presets, slugPrefix ) {
```

Comment:
The logic in the function looks good to me - find the highest preset and increment it by one, nice work putting that together and adding the associated tests.

I think the docblock could be a little more descriptive about what the function does. It mentions 'new index', but the reader has no idea what kind of value that new index will be. It should be possible to understand exactly what the function does without reading any of the code.

Reply:
Acknowledged! I've updated the docblock to provide a clearer description of the function's purpose and its return value.